src/main/webapp/app/iris/iris-course-chat.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@angular/core';
+import { IrisChatHttpService } from 'app/iris/iris-chat-http.service';
+import { IrisWebsocketService } from 'app/iris/iris-websocket.service';
+import { IrisStatusService } from 'app/iris/iris-status.service';
+import { UserService } from 'app/core/user/user.service';
+import { IrisChatService } from 'app/iris/iris-chat-base.service';
+import { AccountService } from 'app/core/auth/account.service';
+
+const IDENTIFIER = 'course-chat/';
+
+@Injectable({ providedIn: 'root' })
+export class IrisCourseChatService extends IrisChatService {
+    courseId?: number;
+
```

Comment:
This component seems to be a duplicate with `IrisExerciseChatService` - I would propose to pass the Ressource (`course-chat/` vs `exercise-chat/`) 

Am I missing something why we can't generalize these components?

Reply:
We probably could